docs/it/testing.md

```diff
@@ -167,6 +167,19 @@ Feature: Google Searching
     Then results for "panda" are shown
 ```
 
+### Mutation Testing
+
+Il mutation testing è una metodologia di test per valutare l'attuale test suite. Partendo dal codice, si creano diverse versioni diverse, ognuna delle quali introduce una mutazione, piccola a piacere. Queste versioni mutate vengono chiamati _mutanti_. L'obiettivo del mutation testing è di verificare se la test suite identifica la mutazione: se un mutante non viene rilevato da un test, viene considerato _vivo_ e i test a riguardo vengono considerati non sufficienti per quella mutazione. Se un mutante viene rilevato da un test, viene considerato _morto_ e i test a riguardo vengono considerati sufficienti per quel mutante.
+
+Applicando la formula `mutation score = (mutanti morti / mutanti totali) * 100`, possiamo ottenere una percentuale di mutanti morti, che ci permette di capire quanto il nostro codice sia realmente testato.
+
+Le mutazioni possono essere effettuate su vari aspetti del proprio software:
+
+- Operazioni (cambiare un "-" con un "+")
+- Comparazioni (cambiare un "==" con un "!=")
+- Istruzioni (eliminare delle righe di codice)
+- Valori (cambiare il _return_ di una funzione)
+
```

Comment:
Ho invertito i 2 paragrafi in modo da dedicare la fine di questo paragrafo allo score.

```suggestion
Le mutazioni possono essere effettuate su vari aspetti del proprio software:
- Operazioni (cambiare un "-" con un "+")
- Comparazioni (cambiare un "==" con un "!=")
- Istruzioni (eliminare delle righe di codice)
- Valori (cambiare il _return_ di una funzione)

Applicando la formula `mutation score = (mutanti morti / mutanti totali) * 100`, possiamo ottenere una percentuale di mutanti morti, che ci permette di capire quanto il nostro codice sia realmente testato.

Vista la natura dei mutation test, il numero di mutazioni può essere molto alto e non sempre le mutazioni coinvolgono i principali use case. Per questo non bisogna stupirsi che lo score sia basso. Infatti uno score alto significa che esiste un test per ogni condizione all'interno del proprio codice, il che porterebbe a:
- un aumento del tempo per la scrittura del codice di test
- un aumento del costo di mantenimento/aggiornamento del codice di test
In base alle esigenze di progetto, il team può decidere se implementare e migliorare il `mutation score`.

Pur essendo un'ottimo strumento per misurare la qualità della test suite, in molti casi questa tecnica di testing non è usata per via della sua complessità e costo di implementazione.
```